src/main.c

```diff
@@ -92,7 +108,7 @@ void timer0_isr() __interrupt 1 __using 1
 {
     // display refresh ISR
     // cycle thru digits one at a time
-    uint8_t digit = displaycounter % 4; 
+    uint8_t digit = displaycounter & 3; 
```

Comment:
just went back to look at main.asm that sdcc 3.5.0 produced - already does this optimization under the hood:
```
;	src/main.c:95: uint8_t digit = displaycounter % 4; 
	mov	a,#0x03
	anl	a,_displaycounter
	mov	r7,a
```